test/models/validatable_test.rb

```diff
@@ -64,7 +64,7 @@ class ValidatableTest < ActiveSupport::TestCase
     user = create_user
 
     user.password = ''
-    user.password_confirmation = ''
+    user.password_confirmation = 'x'
```

Comment:
You can always customize password_required? in your app as well as it seems you have a very specific scenario. :)